pkg/collector/corechecks/servicediscovery/module/ignore_proc_test.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// This doesn't need BPF, but it's built with this tag to only run with
+// system-probe tests, otherwise linters show an error for the core agent tests.
+//go:build linux_bpf
+
+package module
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+// TestShouldIgnorePid check cases of ignored and non-ignored services
+func TestShouldIgnorePid(t *testing.T) {
+	testCases := []struct {
+		name   string
+		comm   string
+		envs   string
+		ignore bool
+	}{
+		{
+			name:   "should ignore datadog agent service",
+			comm:   "agent",
+			envs:   "DD_SERVICE=datadog-agent",
+			ignore: true,
+		},
+		{
+			name:   "should not ignore dummy service",
+			comm:   "dummy",
+			envs:   "DD_SERVICE=dummy",
+			ignore: false,
+		},
+	}
+
+	serverBin := buildTestBin(t)
+	serverDir := filepath.Dir(serverBin)
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			t.Cleanup(func() {
+				cancel()
+				os.Remove(test.comm)
+			})
+
+			makeAlias(t, test.comm, serverBin)
+			bin := filepath.Join(serverDir, test.comm)
+			cmd := exec.CommandContext(ctx, bin)
+			cmd.Env = append(os.Environ(), test.envs)
+			err := cmd.Start()
+			require.NoError(t, err)
+			t.Cleanup(func() {
+				_ = cmd.Process.Kill()
+			})
+
+			discovery := newDiscovery()
+			require.NotEmpty(t, discovery)
+
+			proc, err := customNewProcess(int32(cmd.Process.Pid))
+			require.NoError(t, err)
+
+			require.EventuallyWithT(t, func(collect *assert.CollectT) {
+				// wait until the service name becomes available
+				info, err := discovery.getServiceInfo(proc)
+				assert.NoError(collect, err)
+				assert.NotEmpty(collect, info.ddServiceName)
```

Comment:
you're expecting for a certain value (based on the environment variable value), use it

Reply:
this unit test intends to check that Pid has been bypassed, there are other unit test covering service name extraction. I do not think this unit test should check match of service name.

Reply:
then why do you test `ddServiceName`?

Reply:
I've added service name verification